### ISSUE TITLE
EqualsTo support in ubc and MinLen

### ISSUE THREAD

**Issue by jsantino** [kelloggm/checker-framework#36]
using the equal to operator should expand knowledge about a type

ie 
```
    public static void equalToUpper(@LTLengthOf("a") int m, @LTEqLengthOf("a") int r) {
        if ( r == m ) {
		@LTLengthOf("a") int j = r;
	}
    }

    public static void equalToMinLen(@MinLen(2) String m, @MinLen(0) String r) {
        if ( r == m ) {
		@MinLen(2) String j = r;
	}
    }
```

Test Cases:
checker-framework/checker/tests/upperbound/equalTo.java
checker-framework/checker/tests/minlen/equalTo.java

**Comment by kelloggm** [comment]
Joe,

These should definitely work. If they don't, it's a bug.

**Comment by kelloggm** [comment]
closed for UBC in 07b09f909d02b3ba0dbe51e018f89510cb73b7f8; closed for MinLen in 46d99912a679c9ec6468cdd78c06a6c26202e566
